kratos/CMakeLists.txt

```diff
@@ -129,6 +129,11 @@ endif()
 pybind11_add_module(Kratos MODULE THIN_LTO ${KRATOS_PYTHON_SOURCES})
 target_link_libraries(Kratos PRIVATE ${KRATOS_TEST_LIBRARIES} KratosCore)
 
+# Precompile most demanding / widely included headers
+if(KRATOS_USE_PCH MATCHES ON)
+    target_precompile_headers(KratosCore PRIVATE "$<$<COMPILE_LANGUAGE:CXX>:${CMAKE_CURRENT_SOURCE_DIR}/includes/model_part.h>")
```

Comment:
can we easily extend this or proceed with caution?

Could we add all core-headers here?

Reply:
Caution better, the problem is that this gets included everywhere, so all the sources that do not make use of it will pay an overhead.

The trick is to only pch something that is shared by virtually everyone

Reply:
so what about e.g. `define.h` or `array_1d.h`? Those are already included in `model_part.h`, do they need to be added separately?

I guess `kratos_parameters.h` would be another good addition :thinking:

Reply:
It includes everything that hangs from that header, so array_1d and define are included, along with others